src/oncoexporter/cda/cda_factory.py

```diff
@@ -3,17 +3,13 @@
 import os
 
 import pandas as pd
-from .cda_downloader import CdaDownloader
+
 
 class CdaFactory(metaclass=abc.ABCMeta):
     """Superclass for the CDA Factory Classes
 
     Each subclass must implement the to_ga4gh method, which transforms a row of a table from CDA to a GA4GH Message.
     """
-    def __init__(self, overwrite_downloads:bool=False):
```

Comment:
I'm guessing you removed this b/c downloading isn't actually used here? Just curious

Reply:
Adding the download functionality here implies that *every* `CdaFactory` instance needs the download functionality and I do not believe that is the case. What `CdaFactory` (may) need is a way to map ICD-O to NCIT, nothing else. The mapping code should be completely indifferent of the implementation, e.g. if you get the mapping from remote server, from a `dict`, file, database, whatever.

However, here the `__init__` *must* use the `Downloader`, know about some `overwrite_downloads` parameter, execute some arbitrary code (`get_icdo_to_ncit_path()`). All this is unrelated to the functionality, which is *take a row and give me something*.

So it needs to be moved elsewhere, probably to the place where it is used.

If we find ourselves needing to map ICD-O to NCIT at several places, we should create a class for that.

Reply:
Thanks Daniel that makes sense, thanks for clarifying

Reply:
The extent to which we want to apply these code Nazi rules is another question.. :grimacing: :smile_cat: